lib/spack/spack/build_environment.py

```diff
@@ -126,6 +142,22 @@ def __call__(self, *args, **kwargs):
         return super(MakeExecutable, self).__call__(*args, **kwargs)
 
 
+class ConfigureExecutable(Executable):
+
+    def __init__(self, name, arch):
+        super(ConfigureExecutable, self).__init__(name)
+        self.arch = arch
+
+    def __call__(self, *args, **kwargs):
+        if str(self.arch) != spack.architecture.front_end_sys_type():
```

Comment:
This PR also implicitly relies on `spack.architecture.front_end_sys_type` a proposed function from another PR #4832